python31{1,2}Packages.subliminal: 2.1.0 -> 2.2.1, python31{1,2}Packages.enzyme: 0.4.1 -> 0.5.2, python31{1,2}Packages.pysubs2: 1.6.1 -> 1.7.3

abread opened the pull request:
## Description of changes

Updates subliminal from 2.1.0 to 2.2.1 ([changelog](https://github.com/Diaoul/subliminal/releases)), unbreaking it in the process (the JSON decoder errors mentioned in the derivation's comments are no longer present, and I successfully downloads subtitles with it).
This update changes the supported providers (added `opensubtitlescom`, `gestdown`, `tmdb`; removed `thesubdb`, `argenteam`, `shooter`, `legendastv`).
I don't think it warrants inclusion in the release notes because I expect most people to just use the default option of searching all available providers to find subtitles and the removed providers were removed due to going down.

Additionally, this update required updating some dependencies. Thus, this PR also updates:
- enzyme from 0.4.1 to 0.5.2 ([changelog](https://github.com/Diaoul/enzyme/releases)), which seems to only change the supported Python versions (drops support for Python <3.8, adds support for Python 3.12)
- pysubs2 from 1.6.1 to 1.7.3 ([changelog](https://github.com/tkarabela/pysubs2/blob/1.7.3/docs/release-notes.rst)), which seems to not have any breaking changes apart from dropping support for Python <3.8

Furthermore, the new `pysubs2` broke `ffsubsync`: it used an old `versioneer.py` that prevented it from detecting its dependencies properly. The fix is applied in this PR: it was [already available upstream](https://github.com/smacke/ffsubsync/commit/de75bdbfe846b3376f8c0bcfe2e5e5db82d7ff20) but unreleased, so this PR applies that patch, which is meant to provide Python 3.12 support.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- **N/A(?)** Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- Tested basic functionality of all binary files (usually in `./result/bin/`)
  * [x] pysubs2
  * [x] subliminal
  * [ ] ffsubsync -- not tested, but not expected to break (minimal dependency update, minimal changes to build process)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - **N/A(?)** (Package updates) Added a release notes entry if the change is major or breaking
  - **N/A** (Module updates) Added a release notes entry if the change is significant
  - **N/A** (Module addition) Added a release notes entry if adding a new NixOS module
- [?] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

doronbehar commented on pkgs/development/python-modules/enzyme/default.nix:
```diff
@@ -2,24 +2,27 @@
+    hash = "sha256-fPd5FI2eZusoOGA+rOFAxTw878i4/l1NWgOl+11Xs8E=";
   };
 
+  propagatedBuildInputs = [ setuptools ];
```
Are you sure this shouldn't be `nativeBuildInputs`?

abread replied:
Ah oops, fixed

doronbehar commented on pkgs/by-name/ff/ffsubsync/package.nix:
```diff
@@ -15,10 +16,12 @@ python3.pkgs.buildPythonApplication rec {
-  nativeBuildInputs = with python3.pkgs; [
-    setuptools
+  patches = [
+    ./python312.patch # currently not applied in any released version
```
Why not `fetchpatch` https://github.com/smacke/ffsubsync/commit/de75bdbfe846b3376f8c0bcfe2e5e5db82d7ff20 ?

abread replied:
Completely forgot it existed, it makes a whole lot more sense here, yes. Fixed.

doronbehar reviewed (changes requested):
Hello and thanks for the PR!

2 general and important comments:

1. Every PR should start with the prefix `python31{1,2}Packages.<pkg>`, similarly to the PR title - this way ofborg will guess automatically what attributes to build.
2. Please add a separate commit for the `nixfmt`ing of each file - unfortunately, `nixfmt` adds a lot of diff noise, and that makes it a bit more difficult to review.

abread committed:
python31{1,2}Packages.enzyme: 0.4.1 -> 0.5.2

Note the license change:

https://github.com/Diaoul/enzyme/pull/37/files#diff-c693279643b8cd5d248172d9c22cb7cf4ed163a3c98c8a3f69c2717edd3eacb7

abread committed:
python31{1,2}Packages.pysubs2: 1.6.1 -> 1.7.3

abread committed:
ffsubsync: nixfmt

abread committed:
ffsubsync: fix build by adding a Python 3.12 patch

abread commented:
I addressed all comments (fixed the specific ones, separated out `nixfmt` executions into separate commits, changed the PR title).

That said, I think there may have been a miscommunication here:
> Every PR should start with the prefix python31{1,2}Packages.<pkg>, similarly to the PR title - this way ofborg will guess automatically what attributes to build.

Did you mean "Every commit message" or is it really just the PR title?  @doronbehar

doronbehar commented:
Did you read this:

https://github.com/NixOS/nixpkgs/pull/333098#pullrequestreview-2226816140

?

doronbehar commented:
> Did you read this:
> 
> [#333098 (review)](https://github.com/NixOS/nixpkgs/pull/333098#pullrequestreview-2226816140)
> 
> ?

Oh sorry, we wrote our messages at the same time :).

doronbehar commented:
> > Every PR should start with the prefix python31{1,2}Packages., similarly to the PR title - this way ofborg will guess automatically what attributes to build.
> 
> Did you mean "Every commit message" or is it really just the PR title? @doronbehar

Oh yes sorry, I meant "every commit message"

doronbehar commented:
> I addressed all comments (fixed the specific ones, separated out `nixfmt` executions into separate commits, changed the PR title).

Also the nixfmt commits should start with the same prefix. Read please:

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

abread commented:
I seem to have missed the commit message section while going through the contributing guidelines. However, it should all be fixed now.

doronbehar committed:
python31{1,2}Packages.subliminal: reformat expression a bit

- Don't use `with lib;` in all of meta scope.
- Group arguments with similar purpose together.
- Fix a small typo.
- Don't use `repo = pname` in src.

doronbehar committed:
python31{1,2}Packages.subliminal: 2.1.0 -> 2.2.1; unbreak

doronbehar commented:
OK so I noticed a few semantic changes I wanted to perform further, and it was a bit hard to explain them all in a comments, in comparison to simply do it, so I pushed a new set of commits to the PR... I will merge when CI is green
